src/main/java/io/github/thebusybiscuit/slimefun4/integrations/VaultIntegration.java

```diff
@@ -0,0 +1,42 @@
+package io.github.thebusybiscuit.slimefun4.integrations;
+
+import com.google.common.base.Preconditions;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import net.milkbowl.vault.economy.Economy;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.plugin.RegisteredServiceProvider;
+
+public class VaultIntegration {
+
+    private static Economy economy;
+
+    static void register(@Nonnull Plugin plugin) {
+        RegisteredServiceProvider<Economy> economyProvider = plugin.getServer().getServicesManager().getRegistration(Economy.class);
```

Comment:
will this always return vaults economy class?
what if another plugin has that class name?

Reply:
Official recommend usage. Also used in my own downstream fork for years without issue.

Reply:
> will this always return vaults economy class? what if another plugin has that class name?

If somebody names a class Economy and it is located inside Vault's package, then it is likely Vault.
Yes, this will always be Vault and this is the proper way to hook into it.